pkg/git/git.go

```diff
@@ -88,28 +89,59 @@ func NewGitClientFromExecutor(executor executor.Executor) *Client {
 	}
 }
 
-// CloneRepo clones a repo specified in the gitURL and checks out the specified branch
-func (git *Client) CloneRepo(cloneDirPrefix string, branch string, gitURL string) (string, error) {
-	cloneDir, err := ioutil.TempDir(os.TempDir(), cloneDirPrefix)
+// CloneOptions are the options for cloning a Git repository
+type CloneOptions struct {
+	URL       string
+	Branch    string
+	Bootstrap bool // create the branch if the repository is empty
```

Comment:
Nit: `CreateBranchIfMissing`? (In order to avoid having a comment, and having something hopefully more explicit? It is admittedly longer/more verbose, but hopefully reduces mental load a bit.)

Reply:
Also, given it looks like we always pass `Bootstrap: true` in production code, isn't it worth to _always_ "bootstrap"? 🤔

Reply:
You don't always create the branch, only if the repo is empty. So the name would need to be something like `CreateBranchIfRepoEmpty` which maybe is too long?

We do set it to false when cloning the profile repo

Reply:
Fair enough 👍